src/Plugin.hs

```diff
@@ -65,21 +66,36 @@ gotoDefinition = withAgda $ do
       vim_command $ "keepjumps normal! " <> T.pack (show buffer_idx) <> "go"
 
 
-reload :: Neovim CornelisEnv ()
-reload = do
-  vim_command "noautocmd w"
-  load
-
-
 doLoad :: CommandArguments -> Neovim CornelisEnv ()
 doLoad = const load
 
+atomicSwapIORef :: IORef a -> a -> IO a
+atomicSwapIORef r x = atomicModifyIORef r (\y -> (x , y))
+
+-- Register for buffer update events as late as possible
+-- because neovim starts up plugins twice for no reason.
+firstLoadInit :: Neovim CornelisEnv ()
+firstLoadInit = do
```

Comment:
A better place to set up this subscription is probably in `ftplugin/agda.vim`

Reply:
When I do this on vim's side, I had the problem that the `call` happens much faster than the plugin starts, so if you start writing as soon as you open the file you may get an error because `CornelisNotifyEdit` is not defined yet. Is there a way to put a placeholder until the actual plugin kicks in?

Reply:
I don't care strongly about this, just trying to limit the weird vim-statefulness we stick into the plugin.

I wonder if we could write a little vim wrapper that calls `CornelisNotifyEdit` if the plugin exists, which we can check via `exists(":CornelisNotifyEdit")`

Reply:
I found how to do this in Vim!